src/switchroot/ostree-mount-util.h

```diff
@@ -106,6 +120,69 @@ read_proc_cmdline_ostree (void)
   return ret;
 }
 
+static inline char*
+gen_ostree_target (const char* prepend)
+{
+  char dir_str[PATH_MAX];
+  int dir_str_len = snprintf(dir_str, sizeof(dir_str), "%s", prepend);
+  int dir_str_cap = PATH_MAX - dir_str_len;
+  int prepend_len = dir_str_len;
+  dir_str_len += snprintf(dir_str + prepend_len, sizeof(dir_str), "%s", "/ostree");
+  dir_str_cap = PATH_MAX - dir_str_len;
+  char* append_from = dir_str + dir_str_len;
+  int appended_len = 0;
```

Comment:
There's a lot of hairy C string parsing and arithmetic here.  It really looks to me like this should simplify down to something more like:

`snprintf(dir_str, sizeof(dir_str), "%s/%s", prepend, ostree)`  ?  Not totally sure.

Reply:
It basically saved me doing an extra strlen, by appending prepend and ostree one by one. Because I need the length of the prepend for the very last line of the function.

I can do one sprintf and a strlen though, no biggie, same thing, more readable I guess! ☺️